### PR TITLE
feat: Group By All

### PR DESCRIPTION
**Pull request by berkaysynnada**
There exists a **GROUP BY** usage with **ALL** keyword in some platforms (https://docs.snowflake.com/en/sql-reference/constructs/group-by#label-group-by-all-columns, https://duckdb.org/docs/sql/query_syntax/groupby.html).

It groups together all the columns in the SELECT statement that aren't enclosed within aggregate functions. This streamlines the syntax by keeping the list of columns in one place and helps avoid errors by ensuring that the level of detail in your SELECT matches the grouping in your GROUP BY clause.

When this PR is merged, Datafusion part will be opened. **Exclude** support will also be on the following PR's.

### CODE REVIEW

**Comment by berkaysynnada** [comment]
Hi @alamb, could you please take a look? The Datafusion part is also set, and once this PR is merged, I'll be able to open its corresponding PR.

**Comment by alamb** [comment]
I will do so -- thanks @berkaysynnada

File: src/ast/query.rs [hunk 1]
```diff
@@ -1218,3 +1223,26 @@ impl fmt::Display for SelectInto {
         write!(f, "INTO{}{}{} {}", temporary, unlogged, table, self.name)
     }
 }
+
+#[derive(Debug, Clone, PartialEq, PartialOrd, Eq, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "visitor", derive(Visit, VisitMut))]
+pub enum GroupByExpr {
+    /// ALL
```
Comment (alamb) [inline src/ast/query.rs:1231]:
```suggestion
    /// ALL syntax of [Snowflake], and [DuckDB]
    ///
    /// [Snowflake]: https://docs.snowflake.com/en/sql-reference/constructs/group-by#label-group-by-all-columns
    /// [DuckDB]:  https://duckdb.org/docs/sql/query_syntax/groupby.html).) 
```

**Review by alamb — APPROVED** [review]
Looks good to me -- thank you @berkaysynnada

**Comment by alamb** [comment]
looks like there was a logical conflict on main. I will fix that

**Comment by alamb** [comment]
Thanks @berkaysynnada !@

**Pull request merged**